src/compiler/checker.ts

```diff
@@ -44942,12 +44943,13 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 return false;
             }
         }
-        if (!isImportEqualsDeclaration(node) && node.assertClause) {
+        if (!isImportEqualsDeclaration(node) && node.attributes) {
+            const diagnostic = node.attributes.token === SyntaxKind.WithKeyword ? Diagnostics.Import_attribute_values_must_be_string_literal_expressions : Diagnostics.Import_assertion_values_must_be_string_literal_expressions;

```

Comment:
I would personally just use the attribute error unconditionally.

Reply:
@DanielRosenwasser Should I remove the _import assertions_ errors, or should it be done after the assert is removed?